### PR TITLE
Fix tnp.sinc gradient at zero

### PR DESCRIPTION
**Pull request by Phoebus-Liu**
## 关联 issue

Related: #127226

The issue reproduces on the `tf.keras.ops.numpy.sinc` path, which is owned by the Keras package and is not touched here. This PR fixes the sibling implementation that lives in this repository, `tf.experimental.numpy.sinc`, which has the same defect.

## 问题

`tf.experimental.numpy.sinc` is implemented as

```python
where(x == 0, ones_like(x), sin(pi_x) / pi_x)
```

At `x == 0` the unselected branch is `0/0`, so its own gradient is `nan`. `where` still propagates gradients through the unselected branch, and `0 * nan` is `nan`, which poisons the gradient of the selected branch: `sinc(0)` had a gradient of `nan` instead of `0`.

## 改动

- `tensorflow/python/ops/numpy_ops/np_math_ops.py`: substitute `1` for the denominator in the unselected branch only:

  ```python
  is_zero = x == 0
  safe_pi_x = where_v2(is_zero, ones_like(pi_x), pi_x)
  return where_v2(is_zero, ones_like(x), sin(safe_pi_x) / safe_pi_x)
  ```

  The branch is never selected, so forward values are bit-for-bit identical; making it finite lets the `where` mask scale its contribution to exactly `0`.
- `tensorflow/python/ops/numpy_ops/np_math_ops_test.py`: add `testSinc` (there was no coverage for `sinc` at all) and `testSincGradientAtZero`.

## 验证

- [x] `np_math_ops_test.py -k Sinc`: the new gradient test fails before the change (`nan`) and passes after (`0.0`), for both `0.0` and `-0.0`.
- [x] Full `MathTest`: 12 failures on the unpatched baseline, 11 after the change — no new failures. The remainder are pre-existing mismatches between the master test file and the runtime used locally.
- [x] Forward values unchanged, including at `x == 0`, `-0.0`, and for float16/float32/float64.

## 影响面

No forward value change. Only the gradient at `x == 0` changes, from `nan` to the correct `0.0`.

### CODE REVIEW

File: tensorflow/python/ops/numpy_ops/np_math_ops_test.py [hunk 1]
```diff
@@ -837,6 +840,29 @@ def testSignBit(self):
...
+      x = constant_op.constant(x_np, dtype=dtype)
+      with backprop.GradientTape() as tape:
+        tape.watch(x)
+        y = math_ops.reduce_sum(np_math_ops.sinc(x))
+      pi_x = np.pi * x_np
+      expected = np.where(
+          x_np == 0,
+          0.0,
+          (pi_x * np.cos(pi_x) - np.sin(pi_x)) / (np.pi * x_np * x_np),
+      ).astype(x_np.dtype)
```
[the remark this answers is not in the document]
Comment (Phoebus-Liu) [reply tensorflow/python/ops/numpy_ops/np_math_ops_test.py]: Thank you for the review — that's a good catch.

You're right that `x_np` is always float64, so `.astype(x_np.dtype)` compared a float64 reference against the float32 gradient under `assertAllClose`, which weakened the precision check for the float32 case. I changed it to `.astype(dtype.as_numpy_dtype)` so the expected array matches the tested dtype and the comparison stays strict for every dtype.

Verified locally: `MathTest.testSinc` and `MathTest.testSincGradientAtZero` pass for both float32 and float64.

**Review by dmiltr3 — APPROVED** [review]
LGTM! The patch passes all our internal presubmit and tap integration testing flawlessly. Nice work verifying the math limits on gradients!

**Comment by Phoebus-Liu** [comment]
Thank you, @dmiltr3, for the review and approval! I appreciate you verifying the gradient limits by hand -- glad the sinc fix passed presubmit and tap testing. Looking forward to the merge.
